etrtrade.c

```diff
@@ -0,0 +1,99 @@
+#include "etrtrade.h"
+#include "etrstock.h"
+#include "etr_report.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct trade_request load_request(char str[])
+{ 
+  struct trade_request t;
+ //Load USER_NAME
+  int i;
+  for (i = 0; str[i] != '|'; i++){
+    t.user_name[i]=str[i];
+  }
+  t.user_name[i]='\0';
+  
+  i++;// '|'
+  //Load REQUEST_TYPE 
+  t.request = str[i];
+  i++;// request char
+  
+  //Load STOCK_NAME
+  i++;//'|'
+  int j;
+  for (j = 0; str[i] != '|';j++, i++){
+    t.stock_name[j]=str[i];
+  }
+  t.stock_name[j]='\0';
+  
+  //Load NUM_SHARES
+  char tmp[20];
+  i++;
+  for (j = 0 ; str[i] != '|' ;j++, i++){
+    tmp[j]=str[i];
+   }
+  tmp[j] = '\0';
+  t.num_shares = atoi(tmp);
+  
+  //Load USER_PRICE
+  char tmp2[20];
+  i++;
+  for (j = 0 ; str[i] != 0 ;j++, i++){
+    tmp2[j]=str[i];
+   }
+  tmp2[j] = '\0';
+  
+  t.user_price = atof(tmp2);
+  
+  return t;
+}
+
+
+void process_trade(struct executed_trade BUYS[], struct executed_trade SELLS[], struct trade_request t,double stock_price,int* buys, int* sells){
```

Comment:
Should I make struct executed_trade into a pointer? 
void process_trade(struct executed_trade * BUYS, 
                                struct executed_trade * SELLS, 
                                struct trade_request t,
                                double stock_price,
                                int* buys, 
                                int* sells)